src/code/models/codap-connect.ts

```diff
@@ -837,6 +837,35 @@ export class CodapConnect {
     }
   }
 
+  public addNewDataContextAndTable() {
+    const newDataContextName = `NewTable${Math.floor(10000000000 * Math.random())}`;
```

Comment:
If you're going for a unique-ish identifier then my personal preference is something like current date/time as ticks.